plugins/jobtype/test/azkaban/jobtype/TestTuningParameterUtils.java

```diff
@@ -0,0 +1,164 @@
+    Assert.assertTrue("Failed to get hadoop properties " + MRJobConfig.IO_SORT_MB,
+        confProperties.containsKey(MRJobConfig.IO_SORT_MB));
+    Assert.assertTrue("Failed to get right value for hadoop properties " + MRJobConfig.IO_SORT_MB,
+        confProperties.get(MRJobConfig.IO_SORT_MB).equals("100"));
+
+    Assert.assertTrue("Failed to get hadoop properties " + MRJobConfig.IO_SORT_FACTOR,
+        confProperties.containsKey(MRJobConfig.IO_SORT_FACTOR));
+    Assert.assertTrue("Failed to get right value for hadoop properties " + MRJobConfig.IO_SORT_FACTOR, confProperties
+        .get(MRJobConfig.IO_SORT_FACTOR).equals("100"));
+
+    Assert.assertTrue("Failed to get hadoop properties " + MRJobConfig.REDUCE_MEMORY_MB,
+        confProperties.containsKey(MRJobConfig.REDUCE_MEMORY_MB));
+    Assert.assertTrue("Failed to get right value for hadoop properties " + MRJobConfig.REDUCE_MEMORY_MB, confProperties
+        .get(MRJobConfig.REDUCE_MEMORY_MB).equals("2048"));
+
+    Assert.assertTrue("Failed to get hadoop properties " + MRJobConfig.MAP_MEMORY_MB,
+        confProperties.containsKey(MRJobConfig.MAP_MEMORY_MB));
+    Assert.assertTrue("Failed to get right value for hadoop properties " + MRJobConfig.MAP_MEMORY_MB, confProperties
+        .get(MRJobConfig.MAP_MEMORY_MB).equals("2048"));
+  }
+
+  @Test
+  public void testGetDefaultJobParameterJSON() throws IOException {
+    Props props = new Props();
+    props.put(HadoopConfigurationInjector.INJECT_PREFIX + MRJobConfig.IO_SORT_MB, "100");
+    props.put(HadoopConfigurationInjector.INJECT_PREFIX + MRJobConfig.IO_SORT_FACTOR, "100");
+    props.put(HadoopConfigurationInjector.INJECT_PREFIX + MRJobConfig.REDUCE_MEMORY_MB, "2048");
+    props.put(HadoopConfigurationInjector.INJECT_PREFIX + MRJobConfig.MAP_MEMORY_MB, "2048");
+
+    String defaultParamJSON = TuningParameterUtils.getDefaultJobParameterJSON(props);
+
+    String expectedJSON =
+        "{\"" + MRJobConfig.IO_SORT_MB + "\":\"100\"," + "\"" + MRJobConfig.MAP_MEMORY_MB + "\":\"2048\"," + "\""
+            + MRJobConfig.REDUCE_MEMORY_MB + "\":\"2048\"," + "\"" + MRJobConfig.IO_SORT_FACTOR + "\":\"100\"}";
+
+    Assert.assertEquals("Wrong JSON return for default job parameter ", defaultParamJSON, expectedJSON);
+  }
+
+  @Test
+  public void testUpdateAutoTuningParameters() throws ClientProtocolException, IOException {
+
+    String expectedJSON =
+        "{\"" + MRJobConfig.IO_SORT_MB + "\":\"300\"," + "\"" + MRJobConfig.MAP_MEMORY_MB + "\":\"1024\"," + "\""
+            + MRJobConfig.REDUCE_MEMORY_MB + "\":\"1024\"," + "\"" + MRJobConfig.IO_SORT_FACTOR + "\":\"10\"}";
+    PowerMockito.mockStatic(HttpClients.class);
+    PowerMockito.mockStatic(EntityUtils.class);
+
+    CloseableHttpClient httpClient = Mockito.mock(CloseableHttpClient.class);
+    PowerMockito.when(HttpClients.createDefault()).thenReturn(httpClient);
+    PowerMockito.when(EntityUtils.toString(Matchers.any(HttpEntity.class))).thenReturn(expectedJSON);
```

Comment:
Rather than mocking out a static method, can you simply have your HTTP client return a `StringEntity` with a string defined by you? See http://grepcode.com/file/repo1.maven.org/maven2/org.apache.httpcomponents/httpcore/4.1/org/apache/http/entity/StringEntity.java Then `EntityUtils` should be able to stringify it properly.

Reply:
Done.